src/tyr/serializers.cc

```diff
@@ -157,6 +157,26 @@ void openlr(const valhalla::Api& api, int route_index, rapidjson::writer_wrapper
   writer.end_array();
 }
 
+void serializeWarnings(const valhalla::Api& api, rapidjson::writer_wrapper_t& writer) {
+  writer.start_array("warnings");
+  for (int i = 0; i < api.info().warnings_size(); i++) {
+    writer(std::to_string(i), api.info().warnings(i).description());
+  }
+  writer.end_array();
```

Comment:
if we just do an array of strings and we want to use codes with the strings in the future then this will be a breaking change. so we have to make the decision now whether or not we want coded warnings.

i see why for deprecation we wouldnt need a code. its a warning that is fairly unimportant and at some unspecified date in the future may disappear. however there are other more important and likely permanent warnings about limiting input parameters (max radius for example) or when we need to do a two pass run of a route because we needed to relax hierarchy constraints. i think having codes for messages makes them obviously way easier to detect on the client side and also doesnt hurt us for the less important messages. thoughts?

Reply:
I agree. For simple deprecation notices a plain description will suffice. 

However, for the more permanent warnings, what format will these codes take, so that they aren't confused with error codes? 

We have error codes already from 100 up to 503. (Not all values in-between are used).

Reply:
right but what about the larger point that for things that are important to know but dont stop the request from being processed, i think its important to use a code to distinguish it aside from the text so that its easier to maintain on the client side. with the current implementation in this pr its just an array of strings. can we make it an array of objects, and inside each object we have a code and a string. then `add_warning` can take the code as a parameter and look up the warning from the full list, similar to how the `valhalla_exception_t` stuff works

Reply:
Yes this is a great reason. 
My question is what format should the codes take so as not to confuse them with error codes or HTTP codes on the client side or is it inconsequential?

Reply:
there's no way of confusion, all those things are different entities. you can just add a `unordered_map` with `{int code, string msg}`, put that in src/worker.cc (or .h, I personally don't really mind), then adapt your `add_warning` code

Reply:
> there's no way of confusion, all those things are different entities. you can just add a `unordered_map` with `{int code, string msg}`, put that in src/worker.cc (or .h, I personally don't really mind), then adapt your `add_warning` code

Yes this is what I am currently implementing.